File: frontend/lib/models/sector.dart
enum DiaSemana {
  segunda,
  terca,
  quarta,
  quinta,
  sexta,
  sabado,
  domingo;

  String get valor {
    switch (this) {
      case DiaSemana.segunda:
        return 'segunda';
      case DiaSemana.terca:
        return 'terca';
      case DiaSemana.quarta:
        return 'quarta';
      case DiaSemana.quinta:
        return 'quinta';
      case DiaSemana.sexta:
        return 'sexta';
      case DiaSemana.sabado:
        return 'sabado';
      case DiaSemana.domingo:
        return 'domingo';
    }
  }

  String get rotulo {
    switch (this) {
      case DiaSemana.segunda:
        return 'Segunda-feira';
      case DiaSemana.terca:
        return 'Terça-feira';
      case DiaSemana.quarta:
        return 'Quarta-feira';
      case DiaSemana.quinta:
        return 'Quinta-feira';
      case DiaSemana.sexta:
        return 'Sexta-feira';
      case DiaSemana.sabado:
        return 'Sábado';
      case DiaSemana.domingo:
        return 'Domingo';
    }
  }

  static DiaSemana? fromValor(String? v) {
    if (v == null) return null;
    return DiaSemana.values.firstWhere((d) => d.valor == v, orElse: () => DiaSemana.sabado);
  }
}

class Sector {
  final String id;
  final String nome;
  final DiaSemana? diaSemana;
  final String? hora;
  final String? local;
  final String? ministerioId;
  final String? ministerioNome;
  final String? responsavelNome;

  Sector({
    required this.id,
    required this.nome,
    this.diaSemana,
    this.hora,
    this.local,
    this.ministerioId,
    this.ministerioNome,
    this.responsavelNome,
  });

  factory Sector.fromJson(Map<String, dynamic> json) => Sector(
        id: json['id'] as String,
        nome: json['nome'] as String,
        diaSemana: DiaSemana.fromValor(json['dia_semana'] as String?),
        hora: json['hora'] as String?,
        local: json['local'] as String?,
        ministerioId: json['ministerio_id'] as String?,
        ministerioNome: json['ministerio_nome'] as String?,
        responsavelNome: json['responsavel_nome'] as String?,
      );
}

File: backend/app/models/sector.py
from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class DiaSemana(str, Enum):
    SEGUNDA = "segunda"
    TERCA = "terca"
    QUARTA = "quarta"
    QUINTA = "quinta"
    SEXTA = "sexta"
    SABADO = "sabado"
    DOMINGO = "domingo"


class SectorCreate(BaseModel):
    nome: str = Field(..., min_length=2, max_length=100, description="Ex: Acolhimento, Acólitos")
    dia_semana: Optional[DiaSemana] = Field(None, description="Só para sectores com encontro regular")
    hora: Optional[str] = Field(None, max_length=20, description="Ex: 18:00")
    local: Optional[str] = Field(None, max_length=100)
    ministerio_id: Optional[str] = Field(None, description="Ministério ao qual este sector pertence")
    responsavel_nome: Optional[str] = Field(None, max_length=150)


class SectorUpdate(BaseModel):
    nome: Optional[str] = Field(None, min_length=2, max_length=100)
    dia_semana: Optional[DiaSemana] = None
    hora: Optional[str] = Field(None, max_length=20)
    local: Optional[str] = Field(None, max_length=100)
    ministerio_id: Optional[str] = None
    responsavel_nome: Optional[str] = Field(None, max_length=150)


class SectorOut(BaseModel):
    id: str
    nome: str
    dia_semana: Optional[DiaSemana] = None
    hora: Optional[str] = None
    local: Optional[str] = None
    ministerio_id: Optional[str] = None
    ministerio_nome: Optional[str] = None
    responsavel_nome: Optional[str] = None
    criado_em: datetime
